src/scene_battle_rpg2k.cpp

```diff
@@ -56,6 +57,11 @@ void Scene_Battle_Rpg2k::Update() {
 	}
 
 	Scene_Battle::Update();
+
+	// If an event that changed status finishes without displaying a message window,
+	// we need this so it can update automatically the status_window
+	if (Game_Battle::GetInterpreter().finished_not_updated)
+		status_window->Refresh();
 }
```

Comment:
I don't understand why we can't just call `status_window->Refresh()` every time? @Ghabry do you know why we can't do that and have to invent some logic like this?

Reply:
Because our text rendering is slow (you get noticable frame drops when redrawing each frame on slow systems).
Though considering all the performance improvements over the years maybe is not that bad anymore, should be profiled.